mm/damon/dbgfs.c

```diff
@@ -588,7 +588,7 @@ static int add_init_region(struct damon_ctx *c,
 		return -EINVAL;
 
 	damon_for_each_target(t, c) {
-		if (t->id == target_id) {
+		if ((unsigned long)pid_vnr((struct pid*)t->id) == target_id) {
```

Comment:
Nice catch, thank you!  But, this would result in making 'paddr' case failure, right?  Could you use `targetid_is_pid()` to check if `id` is supposed to be a pointer to a `struct pid` or not?

Reply:
You are right. I only used damon for vaddr :(  Let me try out the 'paddr' function and  resubmit.

Reply:
Thank you, looking forward!